keps/sig-instrumentation/1933-secret-logging-static-analysis/README.md

```diff
@@ -0,0 +1,649 @@
+as proposed in [KEP-1753](https://github.com/kubernetes/enhancements/pull/1754).
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+`go-flow-levee` is in active development and has not yet reached a `v1.0` release.
+It is, however, being developed with specific interest in application to Kubernetes.
+
+
+### Risks and Mitigations
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+As a blocking test, there is a risk for developer toil in the event of any
+false-positive or test flakiness.
+This can be mitigated by any contributed with `/override` permissions.
+
+Similarly, depending the order in which PRs are tested and merged,
+ it is theoretically possible that a merge would result in a violation reaching master.
+If this is not properly handled, all PRs made at that time could present as
+ failing analysis.
+This can be mitigated by consistent scans of `master` as a baseline.
+
+As this analysis depends on project-specific considerations of what constitutes
+ a secret or a sink, periodic review is required to ensure configuration is kept up-to-date.
+This is mitigated somewhat with a consistent use of field tags,
+ as proposed in [KEP-1753](https://github.com/kubernetes/enhancements/pull/1754),
+ though correct application of field tags would also be subject to periodic review.
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Similar to [KEP-20190605: Metrics Validation and Verification](keps/sig-instrumentation/20190605-metrics-validation-and-verification.md),
+this KEP would introduce a test build target to execute analysis during testing.
+The target will be introduced to `//test/instrumentation`, with configuration data 
+ in `//test/instrumentation/testdata`.
+
+The test itself will consist of a script which invokes `go-flow-levee` static analysis
+on all Kubernetes Go packages reachable from `//:all-srcs`. 
+Any findings by the analysis will constitute a test failure.
+Analysis findings reporting the position of both source and sink,
+ indicating to a developer which log call must be corrected and/or which argument must be sanitized.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
```

Comment:
Is there a particular release we want to target this at yet?

Reply:
Does strictly-testing side change adhere to the Kubernetes release cadence?  I'm inclined to improve testing as soon as possible; do you think a 1.20 alpha target reasonable?

KEP-1753 is targeting alpha in 1.20 to provide semantic tags on fields to redact, which this analysis should leverage as well.

Reply:
Here's the deadlines for 1.20:

```
Tuesday, October 6              Week 4     Enhancements Freeze
Thursday, November 12           Week 9     Code Freeze
```

So the KEP would need to be completed, accepted and tracked by Oct. 6 to begin implementation, which must be fully merged by Nov. 12.